comp/core/workloadmeta/collectors/internal/kubeapiserver/deployments.go

```diff
@@ -93,7 +92,7 @@ func (p deploymentParser) Parse(obj interface{}) workloadmeta.Entity {
 
 	for annotation, languages := range deployment.Annotations {
 		// find a match
-		matches := re.FindStringSubmatch(annotation)
+		matches := languagedetectionUtil.AnnotationRegex.FindStringSubmatch(annotation)
```

Comment:
maybe it can be cleaner if you create a function in the `languagedetectionUtil` package that wrap the call to `AnnotationRegex` and so we can keep in private to the package

something like
```suggestion
		matches := languagedetectionUtil.DoesAnnotationsKeyMatch(annotation)
```

Reply:
Thanks for the suggestion. 🙇 

I don't think `DoesAnnotationKeyMatch` is the best name here because it gives a false impression that the function will return a boolean, which is not the case.

Also, your suggestion drew my attention that we can take this a step further and improve code re-usability, so I added `ExtractContainerFromAnnotationKey` which is useful in multiple parts of the code.

WDYT?

Reply:
agree about the name that proposed, I was definitely missing some idea at this time